.github/workflows/beam_PreCommit_Java_ElasticSearch_IO_Direct.yml

```diff
@@ -22,12 +22,16 @@ on:
     paths:
       - "sdks/java/io/elasticsearch/**"
       - "sdks/java/io/elasticsearch-tests/**"
+      - "sdks/java/io/common/**"
+      - "sdks/java/core/src/main/**"
```

Comment:
the original idea was a specific IO PreCommit won't get triggered on common java change (java/core), and one single "Java IO PreCommit" runs basic IO unit tests for these projects. This include elasticsearch.

If the elasticsearch compatibility test suites (":sdks:java:io:elasticsearch-tests:elasticsearch-tests"-5,6,7,8) are noisy we can just remove them from "javaioPreCommit" and no need to adjust the scope of Java_ElasticSearch_IO_Direct

Reply:
Also, for most IOs (except gcp, kafka, jdbc, etc, which triggers on java core change), build the IO jar and running unit tests costs little time than building java core, which was another reason for this "javaioPrecommit" task.

Reply:
Makes sense. For this PR I got rid of the expanded trigger paths. We could probably do that with many others.

For the second comment, I think the extra cost of building the core SDK again is less important than the flaking (plus with the flake we actually rebuild it many times and cost engineer days). But I agree that we don't even need to run it at all. Post-commit and trigger files catch things soon enough.

Also personally I like having a green/red signal that corresponds to a particular module. But I guess with GHA that is not always practical.

Reply:
yeah, if the CI has sufficient resource and API quota, each IO should have its own test workflow so having its own signal, triggered on both io change and java core change. However we have 20+ IOs and too many action triggered at same time causing issue, and the "javaioprecommit" is a (workaround) solution that practices all IO builds while not for PRs with java core change.